packages/twenty-server/src/database/typeorm/core/migrations/billing/1743577268972-addHasReachedCurrentPeriodCapColumnInBillingSubscriptionItemTable.ts

```diff
@@ -0,0 +1,44 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddHasReachedCurrentPeriodCapColumnInBillingSubscriptionItemTable1743577268972
+  implements MigrationInterface
+{
+  name =
+    'AddHasReachedCurrentPeriodCapColumnInBillingSubscriptionItemTable1743577268972';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "core"."billingSubscriptionItem" ADD "hasReachedCurrentPeriodCap" boolean NOT NULL DEFAULT false`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "core"."billingSubscriptionItem" ALTER COLUMN "stripePriceId" SET NOT NULL`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "core"."billingSubscriptionItem" ALTER COLUMN "stripeSubscriptionItemId" DROP NOT NULL`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "core"."billingSubscriptionItem" DROP CONSTRAINT "UQ_6a989264cab5ee2d4b424e78526"`,
+    );
+    await queryRunner.query(
+      `CREATE UNIQUE INDEX "IndexOnStripeNonNullableSubscriptionItemIdUnique" ON "core"."billingSubscriptionItem" ("stripeSubscriptionItemId") WHERE "stripeSubscriptionItemId" IS NOT NULL`,
```

Comment:
why do we change the format?

Reply:
that's for the unicity constraint I guess, makes sense. Let's double check it does not break in production too